solutions/instance/variables.tf

```diff
@@ -56,3 +56,57 @@ variable "access_tags" {
   description = "A list of access tags to apply to the Object Storage instance created by the module. [Learn more](https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial)."
   default     = []
 }
+
+variable "existing_secrets_manager_instance_crn" {
+  type        = string
+  default     = null
+  description = "The CRN of existing secrets manager to use to create service credential secrets for COS instance."
+}
+
+variable "existing_secrets_manager_endpoint_type" {
+  type        = string
+  default     = "public"
+  description = "The type of endpoint to use for communicating with the Secrets Manager instance. Possible values: `public`, `private`. Ignored if `existing_secrets_manager_crn` is null."
+}
+
+variable "service_credential_secrets" {
+  type = list(object({
+    secret_group_name        = string
+    secret_group_description = optional(string)
+    existing_secret_group    = optional(bool, false)
+    service_credentials = list(object({
+      secret_name                             = string
+      service_credentials_source_service_role = string
+      secret_labels                           = optional(list(string))
+      secret_auto_rotation                    = optional(bool)
+      secret_auto_rotation_unit               = optional(string)
+      secret_auto_rotation_interval           = optional(number)
+      service_credentials_ttl                 = optional(string)
+      service_credential_secret_description   = optional(string)
+
+    }))
+  }))
```

Comment:
This is a complex type within a DA and requires documentation, otherwise how is the DA consumer supposed to know what to enter?

The documentation should probably show two secret groups, one of which contains two secrets to demonstrate how to expand it out.

ref: DA-resource-keys.md in this folder and the associated link in the variable description.

Reply:
addressed